interfaces/apparmor/template.go

```diff
@@ -221,22 +212,19 @@ var templateCommon = `
   #deny /{,var/}run/utmp r,
 
   # java
-  @{PROC}/@{pid}/ r,
-  @{PROC}/@{pid}/fd/ r,
+  @{PROC}/@{pid}/{,fd/} r,
   owner @{PROC}/@{pid}/auxv r,
   @{PROC}/sys/vm/zone_reclaim_mode r,
   /etc/lsb-release r,
-  /sys/devices/**/read_ahead_kb r,
+  /sys/devices/**/{,read_ahead_kb,uevent} r,
   /sys/devices/system/cpu/** r,
   /sys/devices/system/node/node[0-9]*/* r,
-  /sys/kernel/mm/transparent_hugepage/enabled r,
-  /sys/kernel/mm/transparent_hugepage/defrag r,
+  /sys/kernel/mm/transparent_hugepage/{enabled,defrag,hpage_pmd_size} r,
   # NOTE: this leaks running process but java seems to want it (even though it
   # seems to operate ok without it) and SDL apps crash without it. Allow owner
   # match until AppArmor kernel var is available to solve this properly (see
   # LP: #1546825 for details). comm is a subset of cmdline, so allow it too.
-  owner @{PROC}/@{pid}/cmdline r,
-  owner @{PROC}/@{pid}/comm r,
+  owner @{PROC}/@{pid}/c{mdline,omm} r,
```

Comment:
this seems a bit like premature optimisation and makes the rule harder to read IMO - how about just:

```
owner @{PROC}/@{pid}/{cmdline,comm} r,
```